### PR TITLE
Fix command documentation and messages

### PR DESCRIPTION
**Pull request by martinfaucheux**
* fix documentations appearing with  `--help` commands
* adjust message (and error messages) to be more explicit and readable to the users
* for auth login command, make `--method web` default (thus optional to specify)

### CODE REVIEW

File: ggshield/cmd/auth/login.py [hunk 1]
```diff
@@ -90,12 +91,16 @@ def login_cmd(
     sso_url: Optional[str],
 ) -> int:
     """
-    Authenticate to your GitGuardian account.
+    Authenticate with a GitGuardian workspace.
+    A successful authentication results in a personal access token.
+    This token is stored in your config and used to authenticate your future requests
```
Comment (elacgit) [inline ggshield/cmd/auth/login.py:96]:
```suggestion
    This token is stored in your config and used to authenticate your future requests.
```

File: ggshield/cmd/auth/logout.py [hunk 2]
```diff
@@ -23,15 +24,18 @@
...
     """
-    Delete saved authentication details for the specified instance (or default instance if not specified)
-    By default, this will also try to revoke found tokens unless --no-revoke is specified.\n
-    If --all is specified, it will iterate over all instances.
+    Remove authentication for a GitGuardian instance.
+    A successful logout results in the deletion of personal access token stored in the configuration.
+    By default, the token will be revoked unless `--no-revoke` option is specified.
+
+    If not specified, ggshield will logout from the default instance.
+    `--all` option can be used if you want to logout from all your GitGuardian instances
```
Comment (elacgit) [inline ggshield/cmd/auth/logout.py:38]:
```suggestion
    The `--all` option can be used if you want to logout from all your GitGuardian instances.
```

Comment (elacgit) [reply ggshield/cmd/auth/logout.py:38]: I double check my wording

File: ggshield/core/oauth.py [hunk 3]
```diff
@@ -339,6 +331,28 @@ def _check_existing_token(self) -> bool:
...
+            str_date = get_pretty_date(expire_at)
+        else:
+            str_date = "never"
+
+        message = (
+            "Success! You are now authenticated.\n"
+            "The personal access token has been created and stored in your ggshield config.\n\n"
+            f"token name: {self._token_name}\n"
+            f"token expiration date: {str_date}\n\n"
+            'You do not need to run "ggshield auth login" again. Future requests will automatically use the token'
```
Comment (elacgit) [inline ggshield/core/oauth.py:351]:
```suggestion
            'You do not need to run "ggshield auth login" again. Future requests will automatically use the token.'
```

File: tests/cmd/auth/test_login.py [hunk 4]
```diff
@@ -370,12 +370,19 @@ def test_valid_process(
...
         else:
-            message += "expiring on " + get_pretty_date(self._get_expiry_date()) + "\n"
+            str_date = get_pretty_date(self._get_expiry_date())
+
+        message = (
+            "Success! You are now authenticated.\n"
+            "The personal access token has been created and stored in your ggshield config.\n\n"
+            f"token name: {self._generated_token_name}\n"
+            f"token expiration date: {str_date}\n\n"
+            'You do not need to run "ggshield auth login" again. Future requests will automatically use the token\n'
```
Comment (elacgit) [inline tests/cmd/auth/test_login.py:383]:
```suggestion
            'You do not need to run "ggshield auth login" again. Future requests will automatically use the token.\n'
```

**Review by elacgit — APPROVED** [review]
Just adjusting my English wordings.

Aside of that all good

**Review by elacaille18 — APPROVED** [review]
I agree with all what @elacgit said 😂

File: ggshield/cmd/config/config_get.py [hunk 5]
```diff
@@ -22,7 +22,7 @@ def config_get_command(
     ctx: click.Context, field_name: str, instance_url: Optional[str]
 ) -> int:
     """
-    Get the value of the specified parameter.
+    Print the value of the given config key
     If --instance is passed, retrieve the value for this specific instance
```
Comment (agateau-gg) [inline ggshield/cmd/config/config_get.py:26]:
```suggestion
    Print the value of the given configuration key.
    If --instance is passed, retrieve the value for this specific instance.
```

File: ggshield/cmd/config/config_list.py [hunk 6]
```diff
@@ -9,7 +9,7 @@
 @click.pass_context
 def config_list_cmd(ctx: click.Context) -> int:
     """
-    List all auth configurations.
+    Print the list of configuration keys and values
```
Comment (agateau-gg) [inline ggshield/cmd/config/config_list.py:12]:
```suggestion
    Print the list of configuration keys and values.
```

File: ggshield/cmd/config/config_set.py [hunk 7]
```diff
@@ -22,7 +22,7 @@ def config_set_command(
     ctx: click.Context, field_name: str, value: str, instance: Optional[str]
 ) -> int:
     """
-    Set the value of a specific field in the auth config.
+    Update the value of the given config key
```
Comment (agateau-gg) [inline ggshield/cmd/config/config_set.py:25]:
```suggestion
    Update the value of the given configuration key.
```

File: ggshield/cmd/config/config_unset.py [hunk 8]
```diff
@@ -23,7 +23,7 @@ def config_unset_command(
     ctx: click.Context, field_name: str, instance_url: Optional[str], all_: bool
 ) -> int:
     """
-    This command removes a parameter value from the auth config.
+    Remove the value of the given config key
```
Comment (agateau-gg) [inline ggshield/cmd/config/config_unset.py:26]:
```suggestion
    Remove the value of the given configuration key.
```

**Review by agateau-gg — CHANGES_REQUESTED** [review]
Minor remarks: some missing `.` and `config` → `configuration`.

The missing `.` are important because `--help` text joins lines with a space unless there is an extra blank line. This script:

```python
import click

@click.command()
def main():
    """
    line1
    line2

    line3
    """
    pass

if __name__ == "__main__":
    main()
```

Produces this `--help` output:

```
$ python help.py --help
Usage: help.py [OPTIONS]

  line1 line2

  line3

Options:
  --help  Show this message and exit.
```

**Review by agateau-gg — APPROVED** [review]

**Pull request merged**